internal/pkg/template/env.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package template
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// Paths of service cloudformation templates under templates/services/.
+const (
+	// EnvCFTemplatePath is the path where the cloudformation for the environment is written.
+	EnvCFTemplatePath       = "environment/cf.yml"
+	fmtEnvCFSubTemplatePath = "environment/cf/%s.yml"
+)
+
+var (
+	// Template names under "environment/cf/".
+	envCFSubTemplateNames = []string{
+		"cfn-execution-role",
+		"custom-resources",
+		"custom-resources-role",
+		"environment-manager-role",
+		"lambdas",
+		"vpc-resources",
+	}
+)
+
+// EnvOpts holds data that can be provided to enable features in an environment stack template.
+type EnvOpts struct {
+	DNSDelegationLambda       string
+	ACMValidationLambda       string
+	EnableLongARNFormatLambda string
+	ImportVpc                 *ImportVpcOpts
+	VpcConfig                 *AdjustVpcOpts
```

Comment:
Would you do me a favor and separate this into two structs

```golang
type EnvImport struct {
  VPC *ImportVPC
} 

type EnvConfig struct {
  VPC *ConfigVPC
}
```

I just want to make it way easier in the future to add new config/import options.

Reply:
(We can even consider nesting them so we don't have to worry about namespacing them.

Reply:
discussed offline and keep it as it is for now